src/mappings/entities/mapSettlement.ts

```diff
@@ -183,6 +185,23 @@ async function handleInstructionFinalizedEvent(
     legs: instruction.legs,
   });
   await Promise.all([settlement.save(), instruction.save()]);
+
+  if (eventId === EventIdEnum.InstructionExecuted) {
+    await Promise.all(
+      instruction.legs.map(async ({ ticker, amount, from, to }) => {
+        const asset = await getAsset(ticker);
+        const settlementAmount = new BigNumber(amount);
+        const fromHolder = await AssetHolder.get(`${from.did}:${ticker}`);
+        const toHolder = await AssetHolder.get(`${to.did}:${ticker}`);
```

Comment:
These aren't guaranteed to be found. I am thinking we need a findOrCreate wrapper.

As is the tests fail here because of this.